chainerx_cc/chainerx/native/native_float16.cc

```diff
@@ -0,0 +1,255 @@
+#include "chainerx/native/native_float16.h"
+
+#include <cstdint>
+
+namespace chainerx {
+namespace native {
+namespace {
+
+union UnionFloatUint {
+public:
+    UnionFloatUint(float v) : f(v) {}
+    UnionFloatUint(uint32_t v) : i(v) {}
+    float f;
+    uint32_t i;
+};
+
+union UnionDoubleUint {
+public:
+    UnionDoubleUint(double v) : f(v) {}
+    UnionDoubleUint(uint64_t v) : i(v) {}
+    double f;
+    uint64_t i;
+};
+
+// Borrowed from npy_floatbits_to_halfbits
+//
+// See LICENSE.txt of ChainerX.
+uint16_t FloatbitsToHalfbits(uint32_t f) {
+    uint16_t h_sgn = static_cast<uint16_t>((f & 0x80000000u) >> 16);
```

Comment:
Let's use uppercase `U` for suffixes.
https://stackoverflow.com/a/10457391/402996